cli/command/system/version.go

```diff
@@ -89,13 +90,15 @@ type clientVersion struct {
 // information.
 func newClientVersion(contextName string, dockerCli command.Cli) clientVersion {
 	v := clientVersion{
-		Version:   version.Version,
-		GoVersion: runtime.Version(),
-		GitCommit: version.GitCommit,
-		BuildTime: reformatDate(version.BuildTime),
-		Os:        runtime.GOOS,
-		Arch:      arch(),
-		Context:   contextName,
+		Version:           version.Version,
+		APIVersion:        api.DefaultVersion,
```

Comment:
> If no server information is fetched:
>
>    docker info --format='{{ json .ClientInfo }}' | jq .ApiVersion
>    "1.44"
>
>If server information is fetched:
>
>    docker info --format='{{ json .}}' | jq .ClientInfo.ApiVersion
>    "1.43"
>
>An alternative could be to leave the ApiVersion field empty if no negotiation took place.

Hmm not sure about this one.

On one side it's weird for the version to be different depending on the format, on the other side side being empty is also not great.

I think leaving it empty if no negotiation took place would be slightly more useful:
- It's already that way so there's no existing expectation that it's always non-empty
- Empty value is better than misleading value as it's easier to notice
- The CLI APIVersion that's not impacted by the server negotiation is already there in `DefaultAPIVersion` and `docker version --format='{{ json .Client }}' | jq .ApiVersion`